app/code/Magento/ProductAlert/Controller/Unsubscribe/Stock.php

```diff
@@ -21,16 +23,33 @@ class Stock extends UnsubscribeController
     protected $productRepository;
 
     /**
-     * @param \Magento\Framework\App\Action\Context $context
-     * @param \Magento\Customer\Model\Session $customerSession
-     * @param \Magento\Catalog\Api\ProductRepositoryInterface $productRepository
+     * @var Magento\Store\Model\StoreManagerInterface
+     */
+    protected $storeManager;
+
+    /**
+     * @var \Magento\ProductAlert\Model\StockFactory
+     */
+    protected $stockFactory;
+
+    /**
+     * Stock constructor.
+     * @param Context $context
+     * @param CustomerSession $customerSession
+     * @param ProductRepositoryInterface $productRepository
+     * @param StoreManagerInterface $storeManager
+     * @param \Magento\ProductAlert\Model\StockFactory $stockFactory
      */
     public function __construct(
         Context $context,
         CustomerSession $customerSession,
-        ProductRepositoryInterface $productRepository
+        ProductRepositoryInterface $productRepository,
+        StoreManagerInterface $storeManager,
+        \Magento\ProductAlert\Model\StockFactory $stockFactory
     ) {
         $this->productRepository = $productRepository;
+        $this->storeManager = $storeManager ?: \Magento\Framework\App\ObjectManager::getInstance()->get(StoreManagerInterface::class);
+        $this->stockFactory = $stockFactory ?: \Magento\Framework\App\ObjectManager::getInstance()->get(\Magento\ProductAlert\Model\StockFactory::class);
```

Comment:
Please use short name of the class.

Reply:
for `ObjectManager` added shortclass name but can't used for `\Magento\ProductAlert\Model\StockFactory` due to same class name in current stock class.